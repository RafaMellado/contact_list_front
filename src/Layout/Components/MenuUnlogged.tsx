import { Button } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";

export interface MenuUnloggedProps {
  translations: string;
}

export function MenuUnlogged({ translations }: MenuUnloggedProps) {
  const { t } = useTranslation();
  const navigate = useNavigate();

  const goToSignUp = () => {
    navigate("/sign-up");
  };

  return (
    <>
      <Button
        data-testid="menu-unlogged-btn"
        className="me-3"
        variant="primary"
        onClick={goToSignUp}
      >
        {t<string>(`${translations}.signUpBtn`)}
      </Button>
    </>
  );
}
