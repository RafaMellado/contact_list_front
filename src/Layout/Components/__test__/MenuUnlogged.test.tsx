import { MenuUnlogged } from "../MenuUnlogged";
import { act, fireEvent, render } from "@testing-library/react";
import type { MenuUnloggedProps } from "../MenuUnlogged";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe(`<${MenuUnlogged.name} />`, () => {
  const history = createMemoryHistory();

  const menuUnloggedProps: MenuUnloggedProps = {
    translations: "contactHistory",
  };

  const factoryComponent = (props: MenuUnloggedProps = menuUnloggedProps) =>
    render(
      <Router location="/" navigator={history}>
        <MenuUnlogged {...props} />
      </Router>
    );

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });

  test("navigate to sign up when click btn", () => {
    const component = factoryComponent();

    act(() => {
      const element = component.getByTestId("menu-unlogged-btn");

      fireEvent.click(element);
    });

    expect(history.location.pathname).toBe("/sign-up");
  });
});
