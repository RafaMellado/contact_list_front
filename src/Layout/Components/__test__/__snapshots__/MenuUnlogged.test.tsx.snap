// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuUnlogged /> renders correctly the component 1`] = `
<DocumentFragment>
  <button
    class="me-3 btn btn-primary"
    data-testid="menu-unlogged-btn"
    type="button"
  >
    contactHistory.signUpBtn
  </button>
</DocumentFragment>
`;
